### PR TITLE
Add tests to cover deleting phone numbers

### PR DESCRIPTION
**Pull request by erikphansen**
## Description
This adds test coverage to ensure that phone number deletion works correctly

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+  await waitForElementToBeRemoved(phoneNumberInput);
+
+  // the edit phone number button should not exist
+  expect(
+    view.queryByText(new RegExp(`edit.*${numberName}`, 'i'), {
+      selector: 'button',
+    }),
+  ).not.to.exist;
+  // and the phone number should not exist
+  // expect(view.queryByText(emailAddress)).not.to.exist;
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx:97]: We should remove this line

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+  // and the add phone number button should exist
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx:137]: Idem here 😄

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+async function testQuickFailure(numberName) {
+  server.use(...mocks.transactionFailed);
+
+  deletePhoneNumber(numberName);
+
+  // expect an error to be shown
+  const alert = await view.findByTestId('edit-error-alert');
+  expect(alert).to.have.descendant('div.usa-alert-error');
+  // TODO: would be nice to be able to check the contents against a RegExp
+  expect(alert).to.contain.text('We’re sorry. We couldn’t update your');
```
Comment (sanlouise) [inline src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx:175]: Can we check here for `We’re sorry. We couldn’t update your ${numberName}` ?'

**Review by sanlouise — APPROVED** [review]
Looks great! Just a few minor comments 👍

File: src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx (hunk 3 above)
Comment (erikphansen) [reply src/applications/personalization/profile-2/tests/components/personal-information/PersonalInformation.delete-telephone.unit.spec.jsx]: Good call. Revised!

**Pull request merged**